### PR TITLE
Extract and confirm types from @nteract/types to flow declares

### PR DESCRIPTION
**Pull request by rgbkrk**
Corrects messaging, channels, and redux types. They're now project global types to be used anywhere and everywhere.

### CODE REVIEW

**Comment by rgbkrk** [comment]
Since I'm in the midst of other fixes, I think I'll also fix the other types while I'm at it in this PR.

Edit: not the records though

File: flow-typed/channels.js [hunk 1]
```diff
@@ -24,7 +25,9 @@ export type ExecuteMessageContent = {
   stop_on_error: boolean
 };
 
-export type ExecuteRequest = JupyterMessage<
+declare type ExecuteRequest = JupyterMessage<
   "execute_request",
   ExecuteMessageContent
 >;
+
+declare type Channels = rxjs$Subject<JupyterMessage<*, *>>;
```
Comment (theengineear) [inline flow-typed/channels.js:33]: 👍

File: flow-typed/npm/rxjs_v5.0.x.js [hunk 2]
```diff
@@ -638,6 +638,90 @@ declare class rxjs$Observable<+T> {
...
+    op4: rxjs$OperatorFunction<C, D>,
+    op5: rxjs$OperatorFunction<D, E>,
+    op6: rxjs$OperatorFunction<E, F>,
+    op7: rxjs$OperatorFunction<F, G>,
+    op8: rxjs$OperatorFunction<G, H>,
+    op9: rxjs$OperatorFunction<H, I>,
+    op10: rxjs$OperatorFunction<I, J>,
+    op11: rxjs$OperatorFunction<J, K>,
+    op12: rxjs$OperatorFunctionLast<K, L>
+  ): L;
```
Comment (theengineear) [inline flow-typed/npm/rxjs_v5.0.x.js:723]: 🙃

File: packages/core/src/epics/execute.js [hunk 3]
```diff
@@ -146,7 +142,7 @@ export function createExecuteCellStream(
     channels &&
     !(kernel.status === "starting" || kernel.status === "not connected");
 
-  if (!kernelConnected) {
+  if (!kernelConnected || !channels) {
```
Comment (theengineear) [inline packages/core/src/epics/execute.js:145]: 💯 flow caught this?

File: flow-typed/base.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// @flow
+
+declare type PrimitiveImmutable = string | number | boolean | null;
+declare type JSONType = PrimitiveImmutable | JSONObject | JSONArray; // eslint-disable-line no-use-before-define
+declare type JSONObject = { [key: string]: JSONType };
+declare type JSONArray = Array<JSONType>;
```
Comment (theengineear) [inline flow-typed/base.js:6]: I almost wonder if we should namespace these types... is there any scenario where we run into possible collisions?

File: packages/types/src/commutable.js [hunk 5]
```diff
@@ -1,19 +1,11 @@
...
-const Immutable = require("immutable");
-const ImmutableMap = Immutable.Map;
-const ImmutableList = Immutable.List;
-
-type PrimitiveImmutable = string | number | boolean | null;
-
-export type JSONType = PrimitiveImmutable | JSONObject | JSONArray; // eslint-disable-line no-use-before-define
-export type JSONObject = { [key: string]: JSONType };
-export type JSONArray = Array<JSONType>;
+import type { Map, List } from "immutable";
```
Comment (theengineear) [inline packages/types/src/commutable.js:3]: Should we normalize this to `import * as Immutable from "immutable"`?

File: packages/types/src/commutable.js [hunk 6]
```diff
@@ -1,19 +1,11 @@
 // @flow
 
-const Immutable = require("immutable");
-const ImmutableMap = Immutable.Map;
-const ImmutableList = Immutable.List;
```
Comment (theengineear) [inline packages/types/src/commutable.js:5]: 🎉 😁

**Review by theengineear — APPROVED** [review]

File: packages/core/src/epics/execute.js (hunk 3 above)
Comment (rgbkrk) [reply packages/core/src/epics/execute.js:145]: Yup!

File: flow-typed/base.js (hunk 4 above)
Comment (theengineear) [reply flow-typed/base.js:6]: ^^ nvm, these won't ever bleed out of our repo, so it's not an issue.

Comment (rgbkrk) [reply flow-typed/base.js:6]: 🎉

File: packages/types/src/commutable.js (hunk 5 above)
Comment (rgbkrk) [reply packages/types/src/commutable.js:3]: We can't do that with (direct) flow types (there aren't utility types like the React ones here).

**Pull request merged**